src/ScottPlot/Plottable/SignalPlotList.cs

```diff
@@ -0,0 +1,129 @@
+
+        /// <summary>
+        /// Number of values in the list.
+        /// This number will never exceed the Capacity.
+        /// </summary>
+        public int Count => NextPointIndex;
+
+        /// <summary>
+        /// Percentage of the capacity currently filled with data
+        /// </summary>
+        public double FillPercentage => 100.0 * Count / Capacity;
+
+        /// <summary>
+        /// Return the last value in the list (or null if empty)
+        /// </summary>
+        public double? LastY => (Count > 0) ? Ys[NextPointIndex - 1] : null;
+
+        /// <summary>
+        /// Return the time point fot the last value in the list (or null if empty)
+        /// </summary>
+        public double? LastX => (Count > 0) ? (NextPointIndex - 1) / SampleRate + OffsetX : null;
+
+        /// <summary>
+        /// Index in the Ys array where the next data point will be placed.
+        /// </summary>
+        private int NextPointIndex = 0;
+
+        /// <summary>
+        /// When the Ys array must be expanded, increase its length by this number of values.
+        /// </summary>
+        private readonly int CapacityIncriment;
+
+        /// <summary>
+        /// Create a SignalPlotList
+        /// </summary>
+        /// <param name="capacity">Initial capacity of the list. Also the size by which to expand it as needed.</param>
+        public SignalPlotList(int capacity) : base()
+        {
+            CapacityIncriment = capacity;
+            Strategy = new LinearDoubleOnlyMinMaxStrategy();
+            Clear(resetCapacity: true);
+        }
+
+        public override string ToString() =>
+            $"SignalPlotList (label={Label}) with {Count}/{Capacity} points filled ({FillPercentage:N2}%)";
+
+        /// <summary>
+        /// Add a single point to the list.
+        /// The internal data structure will be automatically expanded in memory if needed.
+        /// </summary>
+        /// <param name="y">new Y value</param>
+        public void Add(double y)
+        {
+            if (Count >= Capacity)
+                IncreaseCapacity(Capacity + CapacityIncriment);
+            Ys[NextPointIndex] = y;
+            MaxRenderIndex = NextPointIndex;
+            NextPointIndex += 1;
+        }
+
+        /// <summary>
+        /// Add a multiple points to the list.
+        /// The internal data structure will be automatically expanded in memory if needed.
+        /// </summary>
+        /// <param name="ys">new Y values</param>
+        public void AddRange(double[] ys)
+        {
+            // TODO: this can be optimized
+            foreach (var y in ys)
+                Add(y);
+        }
+
+        /// <summary>
+        /// Add a multiple points to the list.
+        /// The internal data structure will be automatically expanded in memory if needed.
+        /// </summary>
+        /// <param name="ys">new Y values</param>
+        public void AddRange(IEnumerable<double> ys)
```

Comment:
There's a nice little optimization (which you can also see in Microsoft's implementation of the STL, where you can compare the requested size to the size given by your growth function. If your growth function gives enough space for the requested size then you go with that, otherwise you just allocate the requested amount:

https://github.com/microsoft/STL/blob/main/stl/inc/vector#L1683-L1685

You can simply get the requested size by taking the current size and adding the length of the new data. However, this optimization requires casting `ys` to a `double[]` or `List<double>` or similar as `IEnumerable.Count()` is a LINQ extension method that loops over the entire collection to get the count as `IEnumerable` does not provide a `Count` property (so you can use it to create generators).

So unfortunately this optimization requires checking if the `ys` is an array or list or something that has an O(1) count field, and then casting to whichever type to use that field.